irohad/main/iroha_conf_loader.hpp

```diff
@@ -20,55 +20,97 @@
 
 #include <fstream>
 #include <string>
+
 #include <rapidjson/rapidjson.h>
-#include <rapidjson/istreamwrapper.h>
+
 #include "common/assert_config.hpp"
+#include "common/types.hpp"
+#include "model/converters/json_common.hpp"
 
 namespace config_members {
-  const char* BlockStorePath = "block_store_path";
-  const char* ToriiPort = "torii_port";  // TODO: Needs AddPeer.
-  const char* KeyPairPath = "key_pair_path";
-  const char* PgOpt = "pg_opt";
-  const char* RedisHost = "redis_host";
-  const char* RedisPort = "redis_port";
+  auto BlockStorePath = "IROHA_BLOCK_STORE_PATH";
+  auto ToriiPort = "IROHA_TORII_PORT";  // TODO: Needs AddPeer.
+  auto KeyPairPath = "IROHA_KEY_PAIR_PATH";
+  auto PgHost = "IROHA_PG_HOST";
+  auto PgPort = "IROHA_PG_PORT";
+  auto PgUser = "IROHA_PG_USER";
+  auto PgPass = "IROHA_PG_PASS";
+  auto RedisHost = "IROHA_REDIS_HOST";
+  auto RedisPort = "IROHA_REDIS_PORT";
+  auto PeerNumber = "IROHA_PEER_NUMBER";
 }  // namespace config_members
 
 /**
- * parse and assert trusted peers json in `iroha.conf`
- * @param iroha_conf_path
+ * Get the value of given environment variable
+ * @param var - name of environment variable
+ * @return value, if variable is set, nullopt otherwise
+ */
+auto getEnvironmentVariable(const char *var) -> nonstd::optional<const char *> {
+  auto val = std::getenv(var);
+  if (val) {
+    return val;
+  }
+  return nonstd::nullopt;
+}
+
+// TODO common json library in libs directory
+namespace iroha {
+  namespace model {
+    namespace converters {
+      template <>
+      struct Convert<size_t> {
+        template <typename T>
+        auto operator()(T &&x) -> nonstd::optional<size_t> {
+          try {
+            return std::stoull(x);
+          } catch (const std::exception &e) {
+            return nonstd::nullopt;
+          }
+        }
+      };
+    }  // namespace converters
+  }    // namespace model
+}  // namespace iroha
+
+/**
+ * Serialize given environment variable to document
+ * @tparam T - environment variable type
+ * @tparam Convert - transform function type
+ * @param var - environment variable for serialization
+ * @param transform - transform function from string to T
+ * @return
+ */
+template <typename T = rapidjson::Value::StringRefType,
+          typename Convert = iroha::model::converters::Convert<T>>
+auto serializeValue(const char *var, Convert transform = Convert()) {
```

Comment:
what is the reason making JSON from env vars?

Reply:
to store them for future use or transfer into another environment (similar machine, etc.) ?

Reply:
better to wait for @lebdron though

Reply:
It is meaningless. Every machine is configured on its own. Config must be considered as private data, since it contains sensitive information like database user/password/location.